widget/hexwidget/hexwidget.go

```diff
@@ -0,0 +1,240 @@
+// Package hexwidget implements a 7-segment style hexadecimal display
+package hexwidget
+
+import (
+	"image"
+	"image/color"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+)
+
+// segmentLookupTable is used by h.Set() - the i-th index into this table
+// represents the raw value that should be sent to updateSegments to show
+// the value i.
+var segmentLookupTable []uint8 = []uint8{
+	1 << 6,
+	(1<<0 | (1 << 1) | (1 << 2) | (1 << 3) | (1 << 6) | (1 << 7)),
+	(1<<2 | (1 << 5)),
+	(1<<4 | (1 << 5)),
+	(1<<0 | (1 << 3) | (1 << 4)),
+	(1<<1 | (1 << 4)),
+	(1 << 1),
+	(1<<3 | (1 << 4) | (1 << 5) | (1 << 6)),
+	0,
+	(1<<3 | (1 << 4)),
+	(1 << 3),
+	(1<<0 | (1 << 1)),
+	(1<<0 | (1 << 1) | (1 << 2) | (1 << 5)),
+	(1<<0 | (1 << 5)),
+	(1<<1 | (1 << 2)),
+	(1<<1 | (1 << 2) | (1 << 3)),
+}
+
+// size in pixels of the hex widget
```

Comment:
These are not pixel sizes, the Fyne coordinate system is device independent

Reply:
1cb3743